pkg/controller/worker/genericactuator/actuator.go

```diff
@@ -211,3 +280,49 @@ func (a *genericActuator) cleanupMachineSets(ctx context.Context, namespace stri
 	}
 	return nil
 }
+
+// shallowDeleteMachineSets deletes the existing machineSets in a given namespace
+// without waiting machine-controll-manager to do this
+func (a *genericActuator) shallowDeleteAllExistingMachineSets(ctx context.Context, namespace string) error {
+	machineSetList := &machinev1alpha1.MachineSetList{}
+	if err := a.client.List(ctx, machineSetList, client.InNamespace(namespace)); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+
+	for _, machineSet := range machineSetList.Items {
+		machineSetDeepCopy := machineSet.DeepCopy()
+		if err := a.client.Delete(ctx, machineSetDeepCopy); client.IgnoreNotFound(err) != nil {
+			return err
+		}
+		if err := extensionscontroller.DeleteAllFinalizers(ctx, a.client, machineSetDeepCopy); err != nil {
+			return errors.Wrap(err, fmt.Sprintf("Error removing finalizer from MachineSet: %s/%s", machineSet.Namespace, machineSet.Name))
+		}
+	}
+	return nil
+}
+
+// shallowDeleteMachines deletes the existing machines in a given namespace
+// without waiting machine-controll-manager to do this
+func (a *genericActuator) shallowDeleteAllExistingMachines(ctx context.Context, namespace string) error {
```

Comment:
Can we make the `shallowDelete<object>` functions a bit more generic? Atleast for `shallowDeleteAllExistingMachines` and `shallowDeleteAllExistingMachineSets`
You can pass in the object type that should be deleted. I

 haven't thought the others could also be generalised.

Reply:
done